userspace/falco/falco.cpp

```diff
@@ -1091,8 +1098,23 @@ int falco_init(int argc, char **argv)
 		}
 		else
 		{
-			open_t open_cb = [](sinsp* inspector) {
-				inspector->open();
+			open_t open_cb = [&udig](sinsp* inspector)
```

Comment:
Not sure this is the best approach, honestly.

I'd prefer `inspector->open_udig()` having its own lambda and then handling whether it should be called or not (via the `udig` boolean flag) at the upper level.

But this all depends on how we want it to behave/integrate (or not) with the following checks(eg., `disable_syscall`).

I'll discuss this with other maintainers too!

Reply:
Can we also bring this up on the call? Right now everything is tightly coupled in the inspector instance.

I have no strong opinions here - this is Loris code :)